x/src/lint/engine/runner.rs

```diff
@@ -0,0 +1,183 @@
+                    continue;
+                }
+
+                // Load up the content for this file.
+                let content_ctx = file_ctx.load()?;
+
+                for linter in linters_to_run {
+                    let source = content_ctx.source(linter.name());
+                    let mut formatter = LintFormatter::new(source, &mut messages);
+
+                    match linter.run(&content_ctx, &mut formatter)? {
+                        RunStatus::Executed => {
+                            // Yay! Lint ran successfully.
+                        }
+                        RunStatus::Skipped(reason) => {
+                            skipped.push((source, reason));
+                        }
+                    }
+
+                    if self.config.fail_fast && !messages.is_empty() {
+                        // At least one issue was found.
+                        break 'outer;
+                    }
+                }
+            }
+        }
+
+        Ok(LintResults { skipped, messages })
+    }
+
+    // ---
+    // Helper methods
+    // ---
+
+    fn get_file_list(&self) -> Result<Vec<&Path>> {
+        // TODO: abstract out SCM and command-running functionality.
+        let stdout = self.files_stdout()?;
+
+        Ok(stdout
+            .split(|b| b == &0)
+            .filter_map(|s| {
+                // TODO: make global exclusions configurable.
+                if !s.is_empty() && !s.starts_with(b"testsuite/libra-fuzzer/artifacts/") {
+                    // Well, this only works on Unix for now. Would be cool to get it working on
+                    // Windows at some point too, but it needs to be done with some care. See
+                    // https://www.mercurial-scm.org/wiki/EncodingStrategy and
+                    // https://en.wikipedia.org/wiki/Mojibake.
+                    use std::os::unix::ffi::OsStrExt;
+                    let s = OsStr::from_bytes(s);
+                    Some(Path::new(s))
+                } else {
+                    None
+                }
+            })
+            .collect())
+    }
+
+    fn files_stdout(&self) -> Result<&[u8]> {
+        self.files_stdout
+            .get_or_try_init(|| {
+                // TODO: abstract out SCM and command-running functionality.
+                let output = Command::new("git")
+                    .current_dir(self.config.project_root)
+                    // The -z causes files to not be quoted, and to be separated by \0.
+                    .args(&["ls-files", "-z"])
+                    .stderr(Stdio::inherit())
+                    .output()?;
+                if !output.status.success() {
+                    return Err(SystemError::Exec {
+                        cmd: "git ls-files",
+                        status: output.status,
+                    });
+                }
+                Ok(output.stdout)
+            })
+            .map(|stdout| stdout.as_slice())
+    }
+}
+
+#[derive(Debug)]
+#[non_exhaustive]
```

Comment:
I didn't know that non_exhaustive worked on structs, TIL

Reply:
It works on structs, enums and struct-like enum variants =)